pdl_schema.md

```diff
@@ -0,0 +1,1106 @@
+---
+layout: guide
+title: PDL Schema
+permalink: /pdl_schema
+redirect_from: /pdl_syntax
+excerpt: Rest.li PDL Schema.
+---
+
+# PDL Schema
+
+## Contents
+-   [PDL Schema Definition](#pdl-schema-definition)
+-   [Creating a Schema](#creating-a-schema)
+-   [Record Type](#record-type)
+-   [Primitive Types](#primitive-types)
+-   [Enum Type](#enum-type)
+-   [Array Type](#array-type)
+-   [Map Type](#map-type)
+-   [Union Type](#union-type)
+-   [Typerefs](#typerefs)
+-   [Fixed Type](#fixed-type)
+-   [Import](#import)
+-   [Properties](#properties)
+-   [Package](#package)
+-   [Escaping](#escaping)
+
+## PDL Schema Definition
+
+Pegasus is a schema definition and serialization system developed as part of
+the Rest.li framework. It provides [multi-language
+support](https://linkedin.github.io/rest.li/multi_language_compatibility_matrix)
+for services built using Rest.li and handles seemless serialization and
+deserialization of data between server and clients.
+
+PDL is a schema definition language for Pegasus, developed
+as a user friendly and concise format replacement for the older JSON based
+[PDSC schema format](https://linkedin.github.io/rest.li/pdsc_syntax).
+
+## Creating a Schema
+
+Pegasus supports different types of schemas: [Records](#record-type),
+[Primitive types](#primitive-types), [Enums](#enum-type), [Arrays](#array-type),
+[Maps](#map-type), [Unions](#union-type), [Fixed](#fixed-type) and
+[Typerefs](#typerefs). Records, Enums and Typerefs have names (Named schemas)
+and thus can be defined as top-level schemas. Named schemas can specify an
+optional namespace to avoid naming conflict between schemas with same name.
+The name prefixed with the namespace using the dot(.) separator becomes the
+fully qualified name(FQN) of the schema. Named schemas can be
+referenced from other schemas using the fully qualified name.
+
+Each top-level schema should be stored in its own file with a `.pdl` extension.
+Name of the file should match the schema name and the directory structure should
+match the namespace (similar to how Java classes are organized).
+
+The Pegasus code generator implements a resolver that is similar to Java class
+loaders. If there is a reference to a named schema, the code generator will try
+to look for a file in the code generator’s resolver path. The resolver path is
+similar to a Java classpath. The fully qualified name of the named schema will
+be translated to a relative file name. The relative file name is computed by
+replacing dots (“.”) in the fully qualified name by the directory path separator
+(typically “/”) and appending a `.pdl` extension. This relative file name looked
+up using each location in the resolver path until it finds a file that contains
+the named schema.
```

Comment:
nit: I think it looks nicer if you escape references to characters. e.g. "/" -> `/`, dot (.) -> (`.`)